File: spring-boot-data-rest/src/main/java/com/jahnelgroup/rest/auction/PersonRepo.java
package com.jahnelgroup.rest.auction;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

@RepositoryRestResource(path = "/people", collectionResourceRel = "/people")
public interface PersonRepo extends CrudRepository<Person, Long> {

}

File: spring-boot-data-rest-security/src/main/kotlin/com/jahnelgroup/datarestsecurity/person/PersonRepo.kt
package com.jahnelgroup.datarestsecurity.person

import org.springframework.data.repository.CrudRepository
import org.springframework.security.access.prepost.PreAuthorize

@PreAuthorize("hasRole('ROLE_USER')")
interface PersonRepo : CrudRepository<Person, Long>{

    @PreAuthorize("hasRole('ROLE_ADMIN')") override fun delete(entities: MutableIterable<Person>?)
    @PreAuthorize("hasRole('ROLE_ADMIN')") override fun delete(entity: Person?)
    @PreAuthorize("hasRole('ROLE_ADMIN')") override fun delete(id: Long?)
    @PreAuthorize("hasRole('ROLE_ADMIN')") override fun deleteAll()

}